packages/block-editor/src/components/block-settings-menu-controls/index.js

```diff
@@ -28,16 +33,41 @@ const BlockSettingsMenuControlsSlot = ( { fillProps, clientIds = null } ) => {
 		};
 	}, [] );
 
+	const slotProps = { ...fillProps, selectedBlocks };
 	return (
-		<Slot fillProps={ { ...fillProps, selectedBlocks } }>
-			{ ( fills ) =>
-				! isEmpty( fills ) && <MenuGroup>{ fills }</MenuGroup>
-			}
-		</Slot>
+		<>
+			<BlockSettingsMenuControlsSlot { ...slotProps } />
+			{ clientIds.length === 1 && (
+				<BlockSettingsMenuControlsSlot
+					{ ...slotProps }
+					name={ getSlotName( clientIds[ 0 ] ) }
+				/>
+			) }
+		</>
 	);
 };
 
-BlockSettingsMenuControls.Slot = BlockSettingsMenuControlsSlot;
+const BlockSettingsMenuControlsSlot = ( props ) => (
+	<Slot { ...props }>
+		{ ( fills ) => ! isEmpty( fills ) && <MenuGroup>{ fills }</MenuGroup> }
+	</Slot>
+);
+
+export const BlockSettingsMenuControls = ( {
+	forAllBlocks = false,
+	...fillProps
+} ) => {
+	const context = useContext( BlockListBlockContext );
+	if ( ! forAllBlocks ) {
+		// Let's use non-existent ID in case the context is missing
+		const clientId = context?.clientId || 'non-such-id';
```

Comment:
Do we have any assertions available? I think using a block-specific mode outside of block context should result, at a very least, in a clear error message in the console (if not blowing up entirely).